libs/cdk-utils/src/lib/runtime/pickFromProcessEnv.ts

```diff
@@ -0,0 +1,52 @@
+/**
+ * Safely extracts required environment variables with type safety
+ *
+ * Validates that all specified environment variables are present and non-empty,
+ * throwing an error if any are missing. Returns both an object and array
+ * representation of the values for different use cases.
+ *
+ * @param keys - The environment variable names to extract (at least one required)
+ * @returns Object containing both object and array representations of the values
+ * @throws Error if any environment variables are missing or empty
+ *
+ * @example
+ * ```typescript
+ * // Extract required environment variables
+ * const { object, values } = pickFromProcessEnv('API_URL', 'SECRET_KEY', 'DB_HOST');
+ *
+ * // Use as object (type-safe keys)
+ * console.log(object.API_URL);  // string
+ * console.log(object.SECRET_KEY);  // string
+ *
+ * // Use as array (maintains order)
+ * const [apiUrl, secretKey, dbHost] = values;
+ * ```
+ */
+export function pickFromProcessEnv<T extends string>(...keys: [T, ...T[]]) {
+    const errors = [];
+    const entries = [];
+
+    for (const key of keys) {
+        const value = process.env[key as keyof typeof process.env];
+
+        if (!value) {
```

Comment:
Should we explicitly check for `undefined` here instead? `process.env` _should_ hold string or undefined, however I'm wary of a boolean getting added and not passing this check.